base/bundle.cpp

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include "base/bundle.hpp"
+
+#include <functional>
+#include <queue>
+
+#include "base/map_util.hpp"
+#include "base/status.hpp"
+
+namespace principia {
+namespace base {
+
+Bundle::Bundle(int threads) {
+  {
+    std::unique_lock<std::mutex> l(lock_);
+    joining_ = false;
+  }
+  for (int i = 0; i < threads; ++i) {
+    threads_.emplace_back([this]() {
```

Comment:
The threads should only be created on demand.  If the bundle is not used it should cost nothing.